# Estimation And Confidence Intervals

## Background
### In quality control processes, especially when dealing with high-value items, destructive sampling is a necessary but costly method to ensure product quality. The test to determine whether an item meets the quality standards destroys the item, leading to the requirement of small sample sizes due to cost constraints.
## Scenario
### A manufacturer of print-heads for personal computers is interested in estimating the mean durability of their print-heads in terms of the number of characters printed before failure. To assess this, the manufacturer conducts a study on a small sample of print-heads due to the destructive nature of the testing process.
## Data
### A total of 15 print-heads were randomly selected and tested until failure. The durability of each print-head (in millions of characters) was recorded as follows:
### 1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29


### Assignment Tasks
### a. Build 99% Confidence Interval Using Sample Standard Deviation
#### Assuming the sample is representative of the population, construct a 99% confidence interval for the mean number of characters printed before the print-head fails using the sample standard deviation. Explain the steps you take and the rationale behind using the t-distribution for this task.


In [1]:
import numpy as np
import scipy.stats as stats

data = np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29])

n = len(data)  
mean_x = np.mean(data)  
std_x = np.std(data, ddof=1) 

df = n - 1

t_critical = stats.t.ppf(0.995, df)

ME = t_critical * (std_x / np.sqrt(n))

CI_lower = mean_x - ME
CI_upper = mean_x + ME

(mean_x, std_x, ME, (CI_lower, CI_upper))


(1.2386666666666666,
 0.19316412956959936,
 0.1484693282152996,
 (1.090197338451367, 1.3871359948819662))

## Final Results
### Sample Mean = 1.239 million characters
### Sample Standard Deviation = 0.193 million characters
### Margin of Error (ME) = 0.148 million characters
### 99% Confidence Interval = (1.090, 1.387) million characters

### b. Build 99% Confidence Interval Using Known Population Standard Deviation
#### If it were known that the population standard deviation is 0.2 million characters, construct a 99% confidence interval for the mean number of characters printed before failure.

In [2]:
sigma = 0.2

z_critical = stats.norm.ppf(0.995)

ME_known_sigma = z_critical * (sigma / np.sqrt(n))

CI_lower_known_sigma = mean_x - ME_known_sigma
CI_upper_known_sigma = mean_x + ME_known_sigma

(ME_known_sigma, (CI_lower_known_sigma, CI_upper_known_sigma))


(0.13301525327090588, (1.1056514133957607, 1.3716819199375725))

## Final Results
### Margin of Error (ME) = 0.133 million characters
### 99% Confidence Interval = (1.106, 1.372) million characters